exercicios/para-casa/entregas/veronica-paranhos/projeto/Conta/Conta.js

```diff
@@ -0,0 +1,167 @@
+class Conta {
+  #agencia;
+  #conta;
+  #saldo;
+  chavesPix;
+  static listaContas = [];
+
+  constructor(agencia, conta, saldo) {
+    this.#agencia = agencia;
+    this.#conta = conta;
+    this.#saldo = saldo;
+    this.chavesPix = {
+      cpf: undefined,
+      email: undefined,
+      telefone: undefined,
+    };
+
+    Conta.listaContas.push(this);
+  }
+
+  getAgencia() {
+    return this.#agencia;
+  }
+
+  getConta() {
+    return this.#conta;
+  }
+
+  getSaldo() {
+    return this.#saldo;
+  }
+
+  setSaldo(novoSaldo) {
+    this.#saldo = novoSaldo;
+  }
+
+  destruir() {
+    let i = Conta.listaContas.indexOf(this);
+    Conta.listaContas.splice(i, 1);
+  }
+
+  criarConta(agencia, conta, saldo) {
+    if (agencia.length === 4 && conta.length === 5 && saldo > 0) {
+      this.#agencia = agencia;
+      this.#conta = conta;
+      this.#saldo = saldo;
+
+      return "Conta criada com sucesso!";
+    } else {
+      throw new Error("Dados inválidos!");
+    }
+  }
+
+  sacar(valor) {
+    if (valor > 0 && typeof valor === "number") {
```

Comment:
Sugestão: a condição `if (valor > 0 && typeof valor === "number")` poderia se tornar, por exemplo, uma função `checarNumeroPositivo()` e ser chamada dentro de `sacar()` e `depositar()`, uma vez que se repete em ambos os métodos. (preciosismo aqui mas to caçando pêlo em ovo pra achar possíveis refatorações 😅 )

Reply:
Aqui elas fazem coisas diferentes caso o valor seja positivo, então acho que não conseguiria aproveitar nos dois métodos. Ou talvez eu que não esteja vendo como fazer.